src/main/java/mods/railcraft/common/fluids/tanks/StandardTank.java

```diff
@@ -87,9 +94,10 @@ public boolean matchesFilter(FluidStack fluidStack) {
 
     @Override
     public int fillInternal(FluidStack resource, boolean doFill) {
-        if (!matchesFilter(resource))
```

Comment:
This check is not for `fillInternal` according to the superclass' javadoc. It is for `fill` only, and you put it in the wrong spot.